src/System.Text.Json/docs/writable_json_dom_spec.md

```diff
@@ -0,0 +1,253 @@
+            }
+        }
+    },
+    {
+        "phone numbers", new JsonArray()
+        {
+            "123-456-7890",
+            "123-456-7890" 
+        }
+    }
+};
+```
+
+JSON array can be also initialized easily in various ways which might be useful in different scenarios:
+
+```csharp
+string[] dishes = { "sushi", "pasta", "cucumber soup" };
+IEnumerable<string> sports = sportsExperienceYears.Where(sport => ((JsonNumber)sport.Value).GetInt32() > 2).Select(sport => sport.Key);
+
+var preferences = new JsonObject()
+{
+    { "colours", new JsonArray { "red", "green", "purple" } },
+    { "numbers", new JsonArray { 4, 123, 88 } },
+    { "prime numbers", new JsonNumber[] { 19, 37 } },
+    { "dishes", new JsonArray(dishes) },
+    { "sports", new JsonArray(sports) },
+    { "strange words", strangeWords.Where(word => ((JsonString)word).Value.Length < 10) },
+};
+```
+
+### Modifying existing instance
+
+The main goal of the new API is to allow users to modify existing instance of `JsonNode` which is not possible with `JsonElement` and `JsonDocument`. 
+
+One may change the existing property to have a different value:
+```csharp
+ var options = new JsonObject { { "use caching", true } };
+ options["use caching"] = (JsonBoolean)false;
+```
+
+Add a value to existing JSON array or property to existing JSON object:
+```csharp
+var bestEmployees = new JsonObject(EmployeesDatabase.GetTenBestEmployees());
+bestEmployees.Add(EmployeesDatabase.GetManager());
+```
+
+Or modify the exisitng property name:
+```csharp
+JsonObject manager = EmployeesDatabase.GetManager();
+JsonObject reportingEmployees = manager.GetJsonObjectProperty("reporting employees");
+reportingEmployees.ModifyPropertyName("software developers", "software engineers");
+```
+
+### Transforming to and from JsonElement
+
+The API allows users to get a writable version of JSON document from a readonly one and vice versa:
+
+Transforming JsonNode to JsonElement:
+```csharp
+JsonNode employeeDataToSend = EmployeesDatabase.GetNextEmployee().Value;
+Mailbox.SendEmployeeData(employeeDataToSend.AsJsonElement());
+```
+
+Transforming JsonElement to JsonNode:
+```csharp
+JsonNode receivedEmployeeData = JsonNode.DeepCopy(Mailbox.RetrieveMutableEmployeeData());
+if (receivedEmployeeData is JsonObject employee)
+{
+    employee["name"] = new JsonString("Bob");
+}
+```
+
+### Parsing to JsonNode
+
+If a developer knows they will be modifying an instance, there is an API to parse string right to `JsonNode`, without `JsonDocument` being an intermediary.
+
+```csharp
+string jsonString = @"
+{
+    ""employee1"" : 
+    {
+        ""name"" : ""Ann"",
+        ""surname"" : ""Predictable"",
+        ""age"" : 30,                
+    },
+    ""employee2"" : 
+    {
+        ""name"" : ""Zoe"",
+        ""surname"" : ""Coder"",
+        ""age"" : 24,                
+    }
+}";
+
+JsonObject employees = JsonNode.Parse(jsonString) as JsonObject;
+employees.Add(EmployeesDatabase.GetNextEmployee());
```

Comment:
Similarly here. We are showing `JsonObject.Add( some arg )`. What would the modified JSON look like if we are not specifying a property name in this case?

Should this be something like this (or maybe the property name is "employee`JObject.Count + 1`"):
```suggestion
employees.Add("employee3", EmployeesDatabase.GetNextEmployee());
```

Reply:
Again - `EmployeesDatabase.GetNextEmployee()` from my examples returned `KeyValuePair<string, JsonNode>` and was ment to test key-value pairs, but here without context it might be confusing, so I will change it.

Reply:
Ah, makes sense. I assumed the wrong return type for `GetNextEmployee`.